### ISSUE TITLE
Use Page Model in Layout

### ISSUE THREAD

**Issue by john-t-white** [toddams/RazorLight#123]
I know you discussed this in the issue https://github.com/toddams/RazorLight/issues/4, but I would like to revisit it if you don't mind.  I understand what you mean about a layout can be used by multiple views, which could have different model types, but I am hoping that you could still look at changing the code to follow exactly what MVC does today.  Currently MVC can do this, which is quite helpful for having a base class your models use, which can then be used directly in the layout by setting the model to the base class. MVC will throw an error if the view model can not be cast to the layout model.  Since this is the default behavior of MVC, I don't think it would be to much to ask that everyone using this to generate templates outside of MVC to follow the same guidelines.  Plus, there should be nothing stopping someone from doing what they do today.  Just have the layout inherit from TemplatePage, not TemplatePage<>, without specifying the model, and they should have access to ViewBag.  Also, the model would be there but would return object instead. Thoughts?

**Comment by kkembel** [comment]
Agreed, I've just run into this trying to use the same pattern as I use for regular MVC razor views. To help with a use case... I use FluentEmail with RazorLight, and as an example I've setup a base "EmailBaseModel" that all of my common model properties sit in for my email Layout.cshtml. Then every email view inherits from EmailBaseModel.

In MVC, I can do this with no problem, I specify the EmailBaseModel as the model for my Layout.cshtml, and the engine is able to cast the InheritedEmailModel for my view to EmailBaseModel so that my layout is able to include model properties common to all of my views.  I set this up for use with RazorLight, assuming that it would behave the same, and am only now coming across these threads after reading why my layout model is always null...

**Comment by lucax88x** [comment]
I am right or this is not yet included in the beta1? Thanks

**Comment by fractos** [comment]
Very much hope this is released soon, I'm having to rewrite the topology of an old Antaris-based system in order to work around models not being available in a layout. Primarily because we switch stuff on/off based on whether the user is logged in.

**Comment by chancie86** [comment]
Been a while since the last release. Any ideas when the next one might be?

**Comment by jzabroski** [comment]
Hoping to do it today.
